### ISSUE TITLE
Can't disable password reset in v4.36 (fixed 4.36.1) (configuration key not expected: authentication_backend.password_reset.disable)

### ISSUE THREAD

**Issue by Kami** [authelia/authelia#3621]
### Description

I upgraded my authelia docker container from v4.35 to v4.36 (47e93b66aaec).

In the previous version I used the following config option to disable password reset:

```yaml
authentication_backend:
  disable_reset_password: true
```

After restarting the container, the container failed to start and produced this message:

```
authelia                 | time="2022-06-28T09:39:20+02:00" level=warning msg="Configuration: configuration key 'authentication_backend.disable_reset_password' is deprecated in 4.36.0 and has been replaced by 'authentication_backend.password_reset.disable': this has been automatically mapped for you but you will need to adjust your configuration to remove this message"
authelia                 | time="2022-06-28T09:39:20+02:00" level=error msg="Configuration: configuration key not expected: authentication_backend.password_reset.disable"
authelia                 | time="2022-06-28T09:39:20+02:00" level=fatal msg="Can't continue due to the errors loading the configuration"
```

Even though it claims the key has been auto mapped, the error is still considered as fatal and the container fails to start.

After that, I update config file to use the new format as recommended by the error message:

```yaml
authentication_backend:
  password_reset:
    disable: true
```

But that didn't seem to work either since I got the following error:

```bash
authelia                 | time="2022-06-28T09:41:07+02:00" level=error msg="Configuration: configuration key not expected: authentication_backend.password_reset.disable"
authelia                 | time="2022-06-28T09:41:07+02:00" level=fatal msg="Can't continue due to the errors loading the configuration"
```

For the time being, to allow the container to start, I simply removed this option. I didn't have the time to dig into the authelia code yet, and I wonder if this is a bug or am doing something wrong.

**Comment by Kami** [comment]
EDIT: Sorry, I just went over the commits (e1ee5a5d071fcaf8aa3cbcfce5f6cef718824918) and it looks like this issue has already been resolved in v4.36.1 - #3616.

**Comment by Kami** [comment]
I confirmed that v4.36.1 (bf5c3722195fe22bf7c26fbab39130ed495e35da05adf166ab096e086841f1d7 image digest) indeed solves this issue.

May be a worthwhile to leave this issue open for a bit in case someone encounter the same issues and then close it.

**Comment by JHrenyk** [comment]
Thanks for leaving it up. helped me out.

**Comment by james-d-elliott** [comment]
Yeah we remap them automatically in memory now (the warning technically can be ignored), but we forgot to update the valid key list. Whoops!

**Comment by SVG62** [comment]
similar situation with version 4.36 after update
`time="2022-06-28T12:28:41+03:00" level=warning msg="Configuration: configuration key 'authentication_backend.disable_reset_password' is deprecated in 4.36.0 and has been replaced by 'authentication_backend.password_reset.disable': this has been automatically mapped for you but you will need to adjust your configuration to remove this message"`

**Comment by Kami** [comment]
@SVG62 per comment above, this has been fixed in v4.36.1

**Comment by SVG62** [comment]
update to version 4.36.1 fixed the work
